code-challenges.test.js

```diff
@@ -12,41 +14,134 @@
 
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
+// Psuedo Code:
+// Takes in a string as a parameter.
+// I will define the mapping of the object.
+// call the function with each variable as an argument.
+// Iterate over each charate in the string.
+// return the encoded message as a string.
+
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
+const secretCodeWord1 = "Lackadaisical";
 // Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
+const secretCodeWord2 = "Gobbledygook";
 // Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
+const secretCodeWord3 = "Eccentric";
 // Expected output: "3cc3ntr1c"
 
 // b) Create the function that makes the test pass.
 
+describe("encodedWords", () => {
+  const secretCodeWords1 = "Lackadaisical";
+  const secretCodeWords2 = "Gobbledygook";
+  const secretCodeWords3 = "Eccentric";
+
+  it("should convert 'a' to 4, 'e' to 3, 'I' to 1, and 'o' to 0", () => {
+    // expect(encodedWords(secretCodeWords1)).toEqual("L4ck4d41s1c4l");
+    // expect(encodedWords(secretCodeWords2)).toEqual("G0bbl3dyg00k");
+    // expect(encodedWords(secretCodeWords3)).toEqual("3cc3ntr1c");
+  });
+  it("should handle empty strings", () => {});
+});
+
+const hiddenMessage = (message) => {
+  message = message.replace(/a/g, "4");
+  message = message.replace(/e/g, "3");
+  message = message.replace(/I/g, "1");
+  message = message.replace(/o/g, "0");
+  return message;
+  return message.replace(/[aeio]/g, (c) => ({ a: 4, e: 3, I: 1, o: 0 }[c]));
+};
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
+// Pseudo Code
+// Create a function that takes in an array of words and a single letter.
+// returns an array of all the words containing that specific letter
+//  attempt to use the .filter method on the input array
+//  will use the .includes metyhod to check if it contains letter
+
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"];
 
-const letterA = "a"
+const letterA = "a";
 // Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
+const letterE = "e";
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
-// b) Create the function that makes the test pass.
+describe("filterWordsByLetter", () => {
```

Comment:
I would encourage you to use a template for your testing and replace the necessary pieces when you are doing a test for a new challenge.